packages/fx-core/src/common/globalState.ts

```diff
@@ -14,16 +17,37 @@ const GlobalStateFileName = "state.json";
  * @param key A string.
  * @return The stored value or `undefined`.
  */
-export function globalStateGet(key: string, defaultValue?: any): any {
+export async function globalStateGet(key: string, defaultValue?: any): Promise<any> {
   const filePath = getGlobalStateFile();
   ensureGlobalStateFileExists(filePath);
 
-  const config = fs.readJSONSync(filePath);
-  let value = config[key];
-  if (value === undefined) {
-    value = defaultValue;
+  const lockFileDir = getLockFolder(filePath);
+  const lockfilePath = path.join(lockFileDir, `${ConfigFolderName}.lock`);
+  await fs.ensureDir(lockFileDir);
+
+  const retryNum = 10;
```

Comment:
Could consider leveraging `retries` option when calling `lock()` later.

Reply:
Will refine in dev branch later,  thanks!